compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3577,6 +3577,128 @@ TR::Register *OMR::Power::TreeEvaluator::vsqrtEvaluator(TR::Node *node, TR::Code
      }
    }
 
+static TR::Register *vminDoubleHelper(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+
+   TR::Register *firstReg = cg->evaluate(firstChild);
+   TR::Register *secondReg = cg->evaluate(secondChild);
+
+   TR::Register *resReg = cg->allocateRegister(TR_VRF);
+
+   TR::Register *cmpReg = cg->allocateRegister(TR_VRF);
+   TR::Register *tempA = cg->allocateRegister(TR_VRF);
+   TR::Register *tempB = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(resReg);
+
+   //n1 is NaN if n1 != n1
```

Comment:
I feel like better comments in this area are needed. If I understand things correctly, Java wants the result of NaN if either operand is NaN but the `xvmindp` instruction doesn't do this. So you need to explain this problem and how it is being dealt with. Like this comment sort of tries to explain how searching for NaNs work but it isn't clear to me what `n1` stands for when the comparison is element by element.